### ISSUE TITLE
No intellisense for props, slots etc. with v4

### ISSUE THREAD

**Issue by Sector6759** [primefaces/primevue#5903]
### Describe the bug

In v4, it seems like some types are not defined correctly as I'm getting no Intellisense for props, slots etc.

I see that you used to extend `ClassComponent` before v4
```typescript
declare class Button extends ClassComponent<ButtonProps, ButtonSlots, ButtonEmits> {}
```
Now in v4 you use `DefineComponent`
```typescript
declare const Button: DefineComponent<ButtonProps, ButtonSlots, ButtonEmits>;
```
It seems like something is still missing somewhere in the type parameter chain.

Intellisense in v3
![image](https://github.com/primefaces/primevue/assets/149817326/489770d9-7d34-4e94-98f3-5963c56223c4)
Intellisense in v4
![image](https://github.com/primefaces/primevue/assets/149817326/3592d4e7-c1cc-4880-890c-e424d610cf3b)

### Reproducer

[StackBlitz](https://stackblitz.com/~/github.com/Sector6759/primevue4)

**Edit:** this reproducer now shows the desired behavior because I updated it according to my [comment](https://github.com/primefaces/primevue/issues/5903#issuecomment-2241709437).

### PrimeVue version

4

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

**Comment by nawazishali** [comment]
Facing the same issue with Nuxt 3.

**Comment by Sector6759** [comment]
@mertsincan I added a reproducer link in my original post.

**Comment by matt-sr** [comment]
Hi, I also have no Intellisense for props - only the documentation comments, with Typescript and without. Vite project, installed following Vite instructions, in styled mode. Should be pretty easy to repro, as I made the project yesterday - version 4.0.0-rc.2

**Comment by tomascf** [comment]
Also encountering the same issue with Nuxt 3

**Comment by Fischer96** [comment]
Yep same issue in all editors I use

**Comment by Sector6759** [comment]
@mertsincan What exactly is the [4.x milestone](https://github.com/primefaces/primevue/milestone/97)? There are a lot of issues which have been opened a "long" time ago. Also, there isn't a single closed issue related to this milestone. Is this merely for collecting issues that will eventually be moved to more specific milestones?

**Comment by Proxx07** [comment]
Also some components doesn't have any intellisense for their props 
![img](https://github.com/primefaces/primevue/assets/88980332/e0d87ff5-b2b2-4677-a286-5378cf537a5d)

**Comment by 2234839** [comment]
My project also has this problem, vscode has no relevant type hints at all
![图片](https://github.com/user-attachments/assets/021f7403-6ead-445a-9e3e-678b753bbd98)

**Comment by 2234839** [comment]
I just have created a new test project at https://github.com/2234839/primevue-test
![images](https://github.com/user-attachments/assets/4a34214e-620c-4500-918c-92530d85b403)
Also unable to get the correct type tips, I use Vue.volar plugin
vscode information is as follows:
Version: 1.91.1 (user setup)
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:06:49.809Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by 2234839** [comment]
![图片](https://github.com/user-attachments/assets/c39d9676-dc57-40a1-8e58-167f5f94fbb4)
I've found that the Tree component gets good type hints, and I hope it helps to improve other components

**Comment by 2234839** [comment]
One problem I found was that when I referenced a component in a vue file using the form 'import Tree from' primevue/tree ', I Ctrl-clicked the component and it jumped to 'declare const Tree'. This type of situation is not good
![图片](https://github.com/user-attachments/assets/23129509-3856-4773-a4ec-8b4f3323d7f6)

When I don't have in vue file references, I use CTRL click components will jump to ` Tree: GlobalComponentConstructor < TreeProps TreeSlots, TreeEmits > `, type hinting at this time is perfect
![图片](https://github.com/user-attachments/assets/24915beb-f24e-48ab-bcc5-04e6fa67f81a)

**Comment by 2234839** [comment]
replace `declare const Tree: DefineComponent<TreeProps, TreeSlots, TreeEmits>;` To `declare a const Tree: GlobalComponentConstructor < TreeProps TreeSlots, TreeEmits >;` can get type hints when you use import. What's the difference between the two methods
 @mertsincan 

![图片](https://github.com/user-attachments/assets/8aaf646d-6f2b-4692-b2b1-4d766d4ef3d4)

**Comment by Sector6759** [comment]
@nawazishali @matt-sr @tomascf @Fischer96 @Proxx07 @2234839

I recently noticed that I wasn't importing PrimeVue components like intended. I always imported the PrimeVue components in my own components, like I did in the example picture, but the docs never say that we should do so. Instead they instruct us to [register](https://primevue.org/vite/#usage) components, by using `app.component()`. To make PrimeVue v4 work, we also need to install the PrimeVue [plugin](https://primevue.org/vite/#plugin) with `app.use()`. After removing the import statement in my own component, intellisense now works as expected. By the way, this was already the intended way in v3 ([register](https://v3.primevue.org/vite/#usage), [plugin](https://v3.primevue.org/vite/#plugin)), but indeed some things behave differently in v4 compared to v3, because in v3, importing a component like I did in my example did not break intellisense.

@mertsincan I'm not sure if this issue can be closed. I assume something's still off, because I don't expect intellisense to break only because of importing PrimeVue components inside my own components.

**Comment by rogrile** [comment]
Same issue here, would love to get back to primevue, but without intellisense the dev time will definitely increase

**Comment by Sector6759** [comment]
@rogrile Did you even read my last comment?

**Comment by rogrile** [comment]
I'm using Nuxt (with a nuxt module for primevue). I have autoimports on, thus I do not import the components myself, nor do I use app.component() or app.use() for plugins. This does not work out of the box, and it does work with primevue 3

**Comment by Sector6759** [comment]
I see. I'd like to get an update on this from the PrimeVue team myself. My only guess is that they are busy fixing all kinds of things.

**Comment by pwang2** [comment]
I can resolve my case with workaround from @Sector6759 mentioned [above](https://github.com/primefaces/primevue/issues/5903#issuecomment-2241709437). Thanks for the information.

TLDR:  If we've already imported primevue using the app.use in `main.ts`, when we import the component again as local Components in SFC, the code IntelliSense will be messed up.  

Even it might be handy, I am strongly not in favor of using a global registration.  Not to mention code actions usually import the component to local imports as we type automatically.  Hope we can hear back from PrimeVue team soon.

**Comment by Sector6759** [comment]
@pwang2 Alternatively you can use [auto imports](https://primevue.org/autoimport/).

**Comment by pwang2** [comment]
> @pwang2 Alternatively you can use [auto imports](https://primevue.org/autoimport/).

Thanks. Yes.   I keep thinking about from a modular even with SFC.   So, I am feeling sort of mixed when using something not imported.   I could be wrong/stubborn here. as `defineProps`/`defineEmits` macro stuff also make the code we author.

**Comment by Eirmas** [comment]
Here is a temporary fix that I did using the PrimeVue auto-import plugin

Enable auto import as described here: https://primevue.org/autoimport/
Use `dts: false` in the options to disable overrides.
```
    plugins: [
      vue(),
      Components({
        dts: false, // We will create this manually to prevent overrides.
        resolvers: [
          PrimeVueResolver({
            components: { prefix: 'P' },
            directives: { prefix: 'P' },
          }),
        ],
      }),
    ],
```

Create a `components.d.ts` file. Put it in your src directory or make sure it's part of the TypeScript compilation
```ts
import { DefineComponent, GlobalComponentConstructor } from '@primevue/core';

type FixPrimeVue<T> =
  T extends DefineComponent<infer Props, infer Slots, infer Emits, infer Methods>
    ? GlobalComponentConstructor<Props, Slots, Emits, Methods>
    : T;

declare module 'vue' {
  export interface GlobalComponents {
    Button: FixPrimeVue<typeof import('primevue/button')['default']>
    // Add the components you are using here
  }
}
```

Hopefully https://github.com/primefaces/primevue/pull/6088 is merged soon which should fix this.

**Comment by Skstud5** [comment]
I encountered the same problem. Absolutely clean project, PrimeVue version 4.1.1, however, this IntelliSense does not work in WebStorm
![image](https://github.com/user-attachments/assets/35051f0c-c89b-4013-99fd-e85efb83ef61)

but works in VScode
![image](https://github.com/user-attachments/assets/deb3a4c4-5433-424a-be9d-4d13994dcc64)

**Comment by Skstud5** [comment]
Yes, everything is set up.
I will try on Saturday or Sunday, I will try to repeat this again. What will happen in the end - I will write here
